zanata-war/src/test/java/org/zanata/test/CdiUnitRunnerWithParameters.java

```diff
@@ -0,0 +1,238 @@
+        }
+    }
+
+    public static class Factory implements ParametersRunnerFactory {
+        @Override
+        public Runner createRunnerForTestWithParameters(TestWithParameters test)
+                throws InitializationError {
+            return new CdiUnitRunnerWithParameters(test);
+        }
+    }
+
+
+    // TODO once https://github.com/BrynCooke/cdi-unit/pull/81 is accepted and released
+    // we just need this method override instead of the horrible hacks below
+//    @Override
+//    @SuppressWarnings("unchecked")
+//    protected <T> T createTest(Class<T> testClass) throws Exception {
+//        T test = super.createTest(testClass);
+//        return injectParameterFields(test);
+//    }
+
+    // TODO delete below here
+    private Object get(String field) {
+        try {
+            Field f = CdiRunner.class.getDeclaredField(field);
+            f.setAccessible(true);
+            return f.get(this);
+        } catch (IllegalAccessException | NoSuchFieldException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void set(String field, Object value) {
+        try {
+            Field f = CdiRunner.class.getDeclaredField(field);
+            f.setAccessible(true);
+            f.set(this, value);
+        } catch (IllegalAccessException | NoSuchFieldException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Class<?> clazz() {
+        return (Class<?>) get("clazz");
+    }
+    private void weld(Weld weld) {
+        set("weld", weld);
+    }
+    private Weld weld() {
+        return (Weld) get("weld");
+    }
+    private void container(WeldContainer container) {
+        set("container", container);
+    }
+    private WeldContainer container() {
+        return (WeldContainer) get("container");
+    }
+    private Throwable startupException() {
+        return (Throwable) get("startupException");
+    }
+    private void startupException(Throwable e) {
+        set("startupException", e);
+    }
+
+    @SuppressWarnings("ThrowableResultOfMethodCallIgnored")
+    @Override
+    protected Object createTest() throws Exception {
+        try {
+            String version = Formats.version(WeldBootstrap.class.getPackage());
+            if("2.2.8 (Final)".equals(version) || "2.2.7 (Final)".equals(version)) {
+                startupException(new Exception("Weld 2.2.8 and 2.2.7 are not supported. Suggest upgrading to 2.2.9"));
+            }
+            weld(new Weld() {
+//                protected Deployment createDeployment(ResourceLoader resourceLoader, CDI11Bootstrap bootstrap) {
+//                    try {
+//                        return new Weld11TestUrlDeployment(resourceLoader, bootstrap, clazz());
+//                    } catch (IOException e) {
+//                        setStartupException(e);
+//                        throw new RuntimeException(e);
+//                    }
+//                }
```

Comment:
It's just part of the code I had to copy from CdiRunner, which won't
compile without a newer version of Weld/CDI. Keeping it in case we upgrade
Weld before this code is all deleted, eg for EAP 7.

On 4 March 2016 at 15:47, Carlos Munoz <notifications@github.com> wrote:

> In
> zanata-war/src/test/java/org/zanata/test/CdiUnitRunnerWithParameters.java
> <https://github.com/zanata/zanata-server/pull/1106#discussion_r54992507>:
>
> > +    @Override
> > +    protected Object createTest() throws Exception {
> > +        try {
> > +            String version = Formats.version(WeldBootstrap.class.getPackage());
> > +            if("2.2.8 (Final)".equals(version) || "2.2.7 (Final)".equals(version)) {
> > +                startupException(new Exception("Weld 2.2.8 and 2.2.7 are not supported. Suggest upgrading to 2.2.9"));
> > +            }
> > +            weld(new Weld() {
> > +//                protected Deployment createDeployment(ResourceLoader resourceLoader, CDI11Bootstrap bootstrap) {
> > +//                    try {
> > +//                        return new Weld11TestUrlDeployment(resourceLoader, bootstrap, clazz());
> > +//                    } catch (IOException e) {
> > +//                        setStartupException(e);
> > +//                        throw new RuntimeException(e);
> > +//                    }
> > +//                }
>
> Any reason for keeping this comment?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/1106/files#r54992507>.
>



-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat